lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+
+        # Preload options as dict - used by facts
+        self.options = self.transform_list_to_dict(self.module.params.get('options'))
+
+        self.validate_etag_support = validate_etag_support
+
+    def __build_argument_spec(self, additional_arg_spec, validate_etag_support):
+
+        merged_arg_spec = dict()
+        merged_arg_spec.update(self.ONEVIEW_COMMON_ARGS)
+
+        if validate_etag_support:
+            merged_arg_spec.update(self.ONEVIEW_VALIDATE_ETAG_ARGS)
+
+        if additional_arg_spec:
+            merged_arg_spec.update(additional_arg_spec)
+
+        return merged_arg_spec
+
+    def __check_hpe_oneview_sdk(self):
+        if not HAS_HPE_ONEVIEW:
+            self.module.fail_json(msg=self.HPE_ONEVIEW_SDK_REQUIRED)
+        if not HAS_FUTURE:
+            self.module.fail_json(msg=self.FUTURE_PACKAGE_REQUIRED)
+
+    def __create_oneview_client(self):
+        if not self.module.params['config']:
+            self.oneview_client = OneViewClient.from_environment_variables()
+        else:
+            self.oneview_client = OneViewClient.from_json_file(self.module.params['config'])
+
+    def execute_module(self):
+        """
+        Abstract function, must be implemented by the inheritor.
+
+        This method is called from the run method. It should contains the module logic
+
+        Returns:
+            dict:
+                 It must return a dictionary with the attributes for the module result,
+                 such as ansible_facts, msg and changed.
+        """
+        raise HPOneViewException("execute_module not implemented")
+
+    def run(self):
+        """
+        Common implementation of the OneView run modules.
+        It calls the inheritor 'execute_module' function and sends the return to the Ansible.
+        It handles any HPOneViewException in order to signal a failure to Ansible, with a descriptive error message.
+        """
+        try:
+            if self.validate_etag_support:
+                if not self.module.params.get('validate_etag'):
+                    self.oneview_client.connection.disable_etag_validation()
+
+            result = self.execute_module()
+
+            if "changed" not in result:
+                result['changed'] = False
+
+            self.module.exit_json(**result)
+
+        except HPOneViewException as exception:
+            self.module.fail_json(msg='; '.join(str(e) for e in exception.args))
+
+    def resource_absent(self, resource, method='delete'):
+        """
+        Generic implementation of the absent state for the OneView resources.
+        It checks if the resource needs to be removed.
+        Args:
+            resource (dict): Resource to delete.
+            method (str):
+                Function of the OneView client that will be called for resource deletion. Usually delete or remove.
+
+        Returns:
+            A dictionary with the expected arguments for the AnsibleModule.exit_json
+
+        """
```

Comment:
We generally use sphinx docstrings which are formatted like this:
```
        """
        Generic implementation of the absent state for the OneView resources.

        It checks if the resource needs to be removed.

        :arg dict resource: Resource to delete.
        :arg str method: Function of the OneView client that will be called for resource deletion. Usually delete or remove.
        :return: A dictionary with the expected arguments for :meth:`ansible.module_utils.basic.AnsibleModule.exit_json`
"""
```
http://www.sphinx-doc.org/en/stable/domains.html#info-field-lists